Documentation/api.md

```diff
@@ -188,6 +192,62 @@ BasicAuth allow an endpoint to authenticate over basic authentication More info:
 
 [Back to TOC](#table-of-contents)
 
+## BlackboxExporterSpec
+
+BlackboxExporterSpec contains specification parameters for the BlackboxExporter used for probing.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| scheme | HTTP scheme to use for scraping. | string | false |
+| url | The URL of the blackbox exporter. | string | true |
+| path | HTTP path to scrape for metrics. | string | false |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitor
+
+BlackboxMonitor defines monitoring for a set of static targets or ingresses.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard object’s metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#objectmeta-v1-meta) | false |
+| spec | Specification of desired Ingress selection for target discovery by Prometheus. | [BlackboxMonitorSpec](#blackboxmonitorspec) | true |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitorList
+
+BlackboxMonitorList is a list of BlackboxMonitors.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard list metadata More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ListMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#listmeta-v1-meta) | false |
+| items | List of BlackboxMonitors | []*[BlackboxMonitor](#blackboxmonitor) | true |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitorSpec
+
+BlackboxMonitorSpec contains specification parameters for a BlackboxMonitor.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| jobLabel | The label to use to retrieve the job name from. | string | false |
+| blackboxExporterSpec | Specification for the blackbox exporter to use for probing targets. | [BlackboxExporterSpec](#blackboxexporterspec) | false |
+| module | The module to use for probing. | string | false |
+| staticTargets | List of static targets for the blackbox exporter. | []string | false |
+| selector | Select Ingress objects by labels. | [metav1.LabelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#labelselector-v1-meta) | false |
+| namespaceSelector | Select Ingress objects by namespace. | [NamespaceSelector](#namespaceselector) | false |
+| params | Optional HTTP URL parameters | map[string][]string | false |
```

Comment:
Do we have existing use cases for this? I would prefer to start with a small API surface and extend once we have legitimate use cases.

Reply:
After checking our configuration again, the sole use case is passing the module that shall be used for probing here, which can already be set via `BlackboxMonitorSpec.Module`. 
So no need to have it around for now.

Reply:
Done